### PR TITLE
Docs: Partner: First pass at documenting endpoints

### PR DESCRIPTION
**Pull request by ebinnion**
JPS v1 required hosting partners perform a lot of integration work setting up oAuth and then integrating in our APIs. We tried to simplify this as much as possible in JPS v2 by shipping two shell scripts that handle provisioning and cancellation of plans.

But, there are cases where a host may actually want to call the API instead of running a CLI command. For example, what if the site is already gone by the time the CLI command can be run?

It's currently pretty straightforward to cancel a plan by making calls directly to the API. A hosting partner simply needs to get an access token then call the cancellation endpoint, passing the domain and access token. That process is what is documented in this PR.

Provisioning a plan via calls directly to the API is much less straightforward. Currently, we require details such as the blog ID to be passed with the plan provisioning call, things that are better handled via the CLI.

For now, provisioning a plan is marked as TBD while we consider how to approach that.

To test:

- Create a partner and key by following instructions here in PCYsg-emq-p2
- Provision a plan manually via Jetpack Start and shell on WPCOM
- Run through commands in this documentation to cancel the plan via the API

### CODE REVIEW

File: docs/partners/plan-provisioning-direct-api.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Query Parameters
+
+- __http_envelope__: Default to `false`. Sending `true` will force the HTTP status code to always be `200`. The JSON response is wrapped in an envelope containing the "real" HTTP status code and headers.
+- __pretty__:        Defaults to `false`. Setting to `true` will output pretty JSON.
+
+#### Response Parameters
+
+- __success__:       (bool) Was the operation successful?.
+- __error_code__:    (string) Error code, if any'.
```
Comment (oskosk) [inline docs/partners/plan-provisioning-direct-api.md]: Looks like this quote is not intended. There's one in the next line too.

**Comment by ebinnion** [comment]
Thanks! I've pushed a commit to remove those quotations.

File: docs/partners/plan-provisioning-direct-api.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+#### Examples
+
+Here is an example using cURL in shell.
+
+```shell
+curl --request POST \
+    --url https://public-api.wordpress.com/oauth2/token \
+    --header 'cache-control: no-cache' \
+    --header 'content-type: multipart/form-data;' \
+    --form client_id={PARTNER_ID} \
```
Comment (gravityrail) [inline docs/partners/plan-provisioning-direct-api.md]: Is this intended to be shell-interpolated? If so, it should probably be written as `${PARTNER_ID}` or `$PARTNER_ID`

File: docs/partners/plan-provisioning-direct-api.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+var options = {
+    method: 'POST',
+    url: 'https://public-api.wordpress.com/oauth2/token',
+    headers: {
+        'cache-control': 'no-cache',
+        'content-type': 'multipart/form-data;'
+    },
+    formData: {
+        client_id: {PARTNER_ID},
```
Comment (gravityrail) [inline docs/partners/plan-provisioning-direct-api.md]: This is also not valid JSON (it should be obvious to most people, but still could be a cause of confusion), so maybe declare these as a `const` above with dummy values and reference them as real variables.

File: docs/partners/plan-provisioning-direct-api.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        client_secret: {PARTNER_SECRET},
+        grant_type: 'client_credentials',
+         scope: 'jetpack-partner'
```
Comment (gravityrail) [inline docs/partners/plan-provisioning-direct-api.md]: Extra spaces

File: docs/partners/plan-provisioning-direct-api.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+```
+
+### Provisioning a plan
+
+TBD.
+
+### Cancelling a plan
+
+Cancelling a Jetpack for a given Jetpack site, as long as the partner making the cancellation request is also the partner that provisioned that plan, is fairly straightforward. To move forward, you'll need your partner secret as well as the URL of the site that you'd like to cancel the plan for.
```
Comment (gravityrail) [inline docs/partners/plan-provisioning-direct-api.md]: > Cancelling a Jetpack _plan_

Comment (gravityrail) [reply docs/partners/plan-provisioning-direct-api.md]: This is also awkward wording in general, a bit longer than it needs to be. Maybe something like:

> Plans can be cancelled by making a request using your Partner Token and the URL of the site being cancelled.

File: docs/partners/plan-provisioning-direct-api.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Endpoint Information
+
+- __Method__: POST
+- __URL__:    https://public-api.wordpress.com/rest/v1.3/jpphp/{$site}/partner-cancel
+
+`$site` is the site's domain and path where `/` in the path is replaced with `::`. For example:
```
Comment (gravityrail) [inline docs/partners/plan-provisioning-direct-api.md]: Maybe include an example with two slashes as well, like

`example.com/demo/wp` becomes `example.com::demo::wp`

File: docs/partners/plan-provisioning-direct-api.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+| 403       | invalid_blog          | The blog ID %s is invalid                                                 |
+| 403       | incorrect_partner_key | Subscriptions can only be cancelled by the oAuth client that created them |
+
+### Examples
+
+Here's an example using cURL in shell.
+
+```shell
+curl --request POST \
+    --url https://public-api.wordpress.com/rest/v1.3/jpphp/{SITE_DOMAIN}/partner-cancel \
```
Comment (gravityrail) [inline docs/partners/plan-provisioning-direct-api.md]: Interpolation tweaks as per comment above

File: docs/partners/plan-provisioning-direct-api.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+Here's an example using the request module in Node JS.
+
+```javascript
+var request = require( 'request ');
+
+var options = {
+    method: 'POST',
+    url: 'https://public-api.wordpress.com/rest/v1.3/jpphp/{SITE_DOMAIN}/partner-cancel',
```
Comment (gravityrail) [inline docs/partners/plan-provisioning-direct-api.md]: Interpolation tweaks as per comment above

File: docs/partners/plan-provisioning-direct-api.md (hunk 2 above)
Comment (ebinnion) [reply docs/partners/plan-provisioning-direct-api.md]: Nope. I just meant it as a placeholder so the partner placed their ID there.

File: docs/partners/plan-provisioning-direct-api.md (hunk 3 above)
Comment (ebinnion) [reply docs/partners/plan-provisioning-direct-api.md]: 👍

**Pull request merged**